src/super_gradients/training/dataloaders/dataloaders.py

```diff
@@ -81,14 +82,24 @@ def get_data_loader(config_name: str, dataset_cls: object, train: bool, dataset_
     return dataloader
 
 
-def _process_dataset_params(cfg, dataset_params, train):
-    default_dataset_params = cfg.train_dataset_params if train else cfg.val_dataset_params
-    default_dataset_params = hydra.utils.instantiate(default_dataset_params)
-    for key, val in default_dataset_params.items():
-        if key not in dataset_params.keys() or dataset_params[key] is None:
-            dataset_params[key] = val
+def _process_dataset_params(cfg, dataset_params, train: bool):
+    # Due to some internal details of Hydra, we need to merge the dataset_params with the default dataset params while retaining the
+    # top-level config structure.
+    # We can't do:
+    # >>> default_dataset_params = cfg.train_dataset_params if train else cfg.val_dataset_params
+    # because it will break interpolation down the road.
+    if train:
+        cfg.train_dataset_params = OmegaConf.merge(cfg.train_dataset_params, dataset_params)
+    else:
+        cfg.val_dataset_params = OmegaConf.merge(cfg.val_dataset_params, dataset_params)
```

Comment:
Does this work when `dataset_params` includes already instantiated objects?
e.g. 
``` python
transforms = [DetectionMosaic(...), DetectionRandomAffine(...), ...]
dataloaders.get('some_dataset', dataset_params={'transforms': transforms})
```

Reply:
I haven't tested this scenario, tbh.

Reply:
😢 😢 😢  No it's not
```
```

Reply:
I found the solution, cannot say I'm very proud of it, but it works on all test cases 🙈